protocols/ledger/ledger_test.go

```diff
@@ -0,0 +1,142 @@
+	destination: types.AddressToDestination(common.HexToAddress(`0x760bf27cd45036a6C486802D30B5D90CfFBE31FE`)),
+	privateKey:  common.Hex2Bytes(`62ecd49c4ccb41a70ad46532aed63cf815de15864bc415c87d507afd6a5e8da2`),
+}
+
+func TestCreateLedger(t *testing.T) {
+	ledgerManager := NewLedgerManager()
+	left := outcome.Allocation{Destination: alice.destination, Amount: big.NewInt(3)}
+	right := outcome.Allocation{Destination: bob.destination, Amount: big.NewInt(2)}
+
+	ledger, err := ledgerManager.CreateTestLedger(left, right, &alice.privateKey, 0)
+	if err != nil {
+		t.Error(err)
+	}
+
+	expectedState := state.State{
+		ChainId:           big.NewInt(9001),
+		Participants:      []types.Address{alice.address, bob.address},
+		ChannelNonce:      big.NewInt(0),
+		AppDefinition:     types.Address{},
+		ChallengeDuration: big.NewInt(45),
+		AppData:           []byte{},
+		Outcome: outcome.Exit{outcome.SingleAssetExit{
+			Allocations: outcome.Allocations{left, right},
+		}},
+		TurnNum: 0,
+		IsFinal: false,
+	}
+
+	gotState := ledger.SignedStateForTurnNum[0].State()
+	if diff := cmp.Diff(expectedState, gotState); diff != "" {
+		t.Errorf("TestCreateLedger: ledger state mismatch (-want +got):\n%s", diff)
+	}
+
+	ledger2, _ := ledgerManager.CreateTestLedger(left, right, &alice.privateKey, 0)
+	if ledger2.ChannelNonce.Cmp(big.NewInt(1)) != 0 {
+		t.Error("TestCreateLedger: ledger channel should use the next nonce")
+	}
+
+}
+
+func TestHandleLedgerRequest(t *testing.T) {
+	ledgerManager := NewLedgerManager()
+	left := outcome.Allocation{Destination: alice.destination, Amount: big.NewInt(3)}
+	right := outcome.Allocation{Destination: bob.destination, Amount: big.NewInt(2)}
+
+	ledger, _ := ledgerManager.CreateTestLedger(left, right, &alice.privateKey, 0)
+
+	destination := types.AddressToDestination(common.HexToAddress(`0x5e29E5Ab8EF33F050c7cc10B5a0456D975C5F88d`))
+	asset := types.Address{}
+	oId := protocols.ObjectiveId("Test")
+
+	validRequest := protocols.LedgerRequest{ObjectiveId: oId, LedgerId: ledger.Id, Left: left.Destination, Right: right.Destination, Destination: destination, Amount: types.Funds{asset: big.NewInt(4)}}
+	invalidRequest := protocols.LedgerRequest{ObjectiveId: oId, LedgerId: ledger.Id, Left: left.Destination, Right: right.Destination, Destination: destination, Amount: types.Funds{asset: big.NewInt(10)}}
+
+	_, err := ledgerManager.HandleRequest(ledger, validRequest, &alice.privateKey)
+	if err == nil {
+		t.Errorf("TestHandleLedgerRequest: expected request to be fail as there is no supported state")
+	}
+
+	SignPreAndPostFundingStates(ledger, []*[]byte{&alice.privateKey, &bob.privateKey})
+
+	_, err = ledgerManager.HandleRequest(ledger, invalidRequest, &alice.privateKey)
+	if err == nil {
+		t.Errorf("TestHandleLedgerRequest: expected request to  fail as the ledger does not have enough funds")
+	}
+
+	sideEffects, err := ledgerManager.HandleRequest(ledger, validRequest, &alice.privateKey)
```

Comment:
```suggestion
	_, err = ledgerManager.HandleRequest(ledger, invalidRequest, &alice.privateKey)
	if err == nil {
		t.Errorf("TestHandleLedgerRequest: expected request to  fail as the ledger does not have enough funds")
	}

	SignPreAndPostFundingStates(ledger, []*[]byte{&alice.privateKey, &bob.privateKey})
	sideEffects, err := ledgerManager.HandleRequest(ledger, validRequest, &alice.privateKey)
```

moving the 'fix' closer to the test that the fix was applied.